CHANGELOG/v1.12.0-rc.0.md

```diff
@@ -0,0 +1,394 @@
+🚨 This is a RELEASE CANDIDATE. Use it only for testing purposes. If you find any bugs, file an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/new).
+
+
+:warning: **RELEASE CANDIDATE NOTES** :warning:
+## 👌 Kubernetes version support
+
+- Management Cluster: v1.31.x -> v1.34.x
+- Workload Cluster: v1.29.x -> v1.34.x
+
+[More information about version support can be found here](https://cluster-api.sigs.k8s.io/reference/versions.html)
+
+## Highlights
+
+- Bumped to Go 1.24, controller-runtime v0.22, k8s.io/* v0.34, controller-gen v0.19
```

Comment:
```suggestion
- Bumped to Go 1.24, controller-runtime v0.22, k8s.io/* v0.34, controller-gen v0.19
- In-place updates
    - Can be enabled with the new `InPlaceUpdates` feature gate
    - More details can be found in the [proposal](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240807-in-place-updates.md) 
- Chained upgrades
    - More details can be found in the [proposal](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20250513-chained-and-efficient-upgrades-for-clusters-with-managed-topologies.md)
    - Runtime SDK: Make the AfterClusterUpgrade hook blocking (#12984)
```

Didn't get around to finish the TODOs yet, but would be good if we already have something